app/src/main.rs

```diff
@@ -118,71 +100,67 @@ fn main() {
 
     let mut retval = sgx_status_t::SGX_SUCCESS;
 
-    let binary_median_int = fs::read(WASM_FILE_MEDIAN_INT).unwrap();
-    let binary_median_float = fs::read(WASM_FILE_MEDIAN_FLOAT).unwrap();
-
-    let binary_mean_int = fs::read(WASM_FILE_MEAN_INT).unwrap();
-    let binary_mean_float = fs::read(WASM_FILE_MEAN_FLOAT).unwrap();
-
-    let binary_sd_int = fs::read(WASM_FILE_SD_INT).unwrap();
-    let binary_sd_float = fs::read(WASM_FILE_SD_FLOAT).unwrap();
-
-    let mut result_out_median_int = 0i32;
-    let mut result_out_median_float = 0f32;
-
-    let mut result_out_mean_int = 0i32;
-    let mut result_out_mean_float = 0f32;
-
-    let mut result_out_sd_int = 0f32;
-    let mut result_out_sd_float = 0f32;
+    //// Mean test
+    // Numbers float and integer mixed
+     let data_median = read_data_from_json(
+        "/root/workspace/wasm-exec-sgx/get-median-wasm/test_data.json",
+        "median_int_works_odd",
+    )
+    .unwrap();
+    let serialized_data_median: Vec<u8> = serde_json::to_vec(&data_median).unwrap(); // Create a new byte array that holds the serialized JSON data
+    let binary_median = fs::read(WASM_FILE_MEDIAN).unwrap();
+    let mut result_out_median = 0f32;
+
+    //// Mean test
+    // Numbers float and integer mixed
+    let data_mean = read_data_from_json(
+        "/root/workspace/wasm-exec-sgx/get-mean-wasm/test_data.json",
+        "mean_int_works",
+    )
+    .unwrap();
+    let serialized_data_mean: Vec<u8> = serde_json::to_vec(&data_mean).unwrap(); // Create a new byte array that holds the serialized JSON data
+    let binary_mean = fs::read(WASM_FILE_MEAN).unwrap();
+    let mut result_out_mean = 0f32;
+
+    // Sd test
+    // Numbers float and integer mixed
+    let data_sd = read_data_from_json(
+        "/root/workspace/wasm-exec-sgx/get-sd-wasm/test_data.json",
```

Comment:
The file path needs to change, "/home/bill/workspace..." worked for me for my system.